starter_code/src/movies.js

```diff
@@ -1,20 +1,112 @@
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+// Turn duration of the movies from hours to minutes
+function stringToMinutes(str) {
+  var arr = str.split(" ");
+  if (arr.length === 2) {
+    var hours = parseInt(arr[0]);
+    var minutes = parseInt(arr[1]);
+  } else if (str.includes("m")) {
+    var minutes = parseInt(arr[0]);
+    var hours = 0;
+  } else {
+    var hours = parseInt(arr[0]);
+    var minutes = 0;
+  }
+  return hours * 60 + minutes;
+}
 
+function turnHoursToMinutes(array) {
+  var newMovie = [];
+  var resultDuration = array.map(function(oneMovie) {
+    var newArr = {
+      title: oneMovie.title,
+      year: oneMovie.year,
+      director: oneMovie.director,
+      duration: stringToMinutes(oneMovie.duration),
+      genre: oneMovie.genre,
+      rate: oneMovie.rate
+    };
 
-// Get the average of all rates with 2 decimals 
+    newMovie.push(newArr);
+  });
+  return newMovie;
+}
 
+turnHoursToMinutes(movies);
+
+// Get the average of all rates with 2 decimals
+function ratesAverage(array) {
+  if (array.length === 0) {
+    return undefined;
+  }
+  return Number(
+    (
+      array.reduce(function(sum, rating) {
+        return (sum += Number(rating.rate));
+      }, 0) / array.length
+    ).toFixed(2)
+  );
+}
+
+ratesAverage(movies);
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate(array) {
+  var drama = array.filter(function(movie) {
+    return movie.genre.includes("Drama");
+  });
+  if (drama.length > 0) {
+    return ratesAverage(drama);
+  }
+}
+
+dramaMoviesRate(movies);
 
 // Order by time duration, in growing order
 
+function orderByDuration(array) {
+  return array.sort(function(a, b) {
+    if (Number(a.year) < Number(b.year)) {
+      return -1;
+    } else {
+      return 1;
+    }
+  });
+}
+
```

Comment:
Since you turn years into numbers, it would be simpler to put     Number(a.year) - Number(b.year)  in the call back of your sort method, the way you did it is more used for strings. To compare movies based on their title if they have the same duration, you can jus add a condition : 
```js
function orderByDuration(moviesArray) {
  moviesArray.sort(function(a, b) {
    if (a.duration === b.duration) {
      if (a.title > b.title) {
        return 1;
      } else {
        return -1;
      }
    }
    return a.duration - b.duration;
  });
  return moviesArray;
}
```